app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def update_users_posts_counter
+    user.increment!(:post_counter)
+  end
```

Comment:
- [ ] [OPTIONAL] Please consider using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_users_posts_counter` method every time a new post has been saved to database, then make the `#update_users_posts_counter` method private. (you can do the following to the rest of the `update` methods in other models)